# Characterisation of the genetic determinants of context specific DNA methylation in primary monocytes

**Abstract**
DNA methylation (DNAm) has pervasive effects on gene expression and associations with ageing-related traits. Here we describe monocyte DNAm responses to inflammatory stimuli across 190 individuals. We find that, unlike the similarly widespread changes in gene expression elicited by LPS and IFNÎ³, DNAm is markedly more sensitive to LPS. Exposure to LPS caused differential methylation at 20,858 immune modulated CpGs (imCpGs) which display distinct genomic localisation and transcription factor usage, dependent upon whether methylation is lost or gained. Demethylated imCpGs are profoundly enriched for enhancers, and are over-represented by genes implicated in human dis eases, most notably cancer. We find LPS-induced demethylation follows hydroxymethylation and for most sites the degree of demethylation correlates with baseline signal. Notably, we find LPS exposure triggers gain in epigenetic age by approximately 6 months, identifying a potential cause of accelerated epigentic aging which has diverse negative health associations. Finally, we explore the effect of genetic variation on LPS induced changes in DNAm in 188 individuals, identifying 234 imCpGs under genetic control. Exploring shared causal loci between LPS induced DNAm responses and human disease traits highlights examples of human disease associated loci that also modulate imCpG formation. In summary, our findings suggest innate immune activity continually remodels DNAm in a highly punctate, enhancer enriched fashion that is under tight genetic control and predominantly involves genes commonly mutated in cancer.

[Preprint](https://www.biorxiv.org/content/10.1101/2023.05.17.541041v1)

**Overview of repository**
* LPS-induced differential methylation in monocytes: contains R script and source data to reproduce analysis and figures exploring changes in DNAm in monocytes in response to LPS stimulation.
